### ISSUE TITLE
Recursive import_loop with ListType

### ISSUE THREAD

**Issue by Guibod** [schematics/schematics#453]
I am unable to recursively apply update on an object:

```
model_instance.import_data(request.get_json(force=True), recursive=True)
```

As one of my subobject is a ListType and that it crashes on this very line:

```
data = dict(context.trusted_data) if context.trusted_data else {}
```

My straight array can't be cast to dict, I guess it would work with a 0 or 2 element list.

Attached my test case proving my point:

```
class Stuff(Model):
    label = StringType()


class MyOtherModel(Model):
    name = StringType(default='fu')
    list = ListType(ModelType(Stuff))


class TestListImport(unittest.TestCase):
    def test_model_import(self):
        instance = MyOtherModel()
        instance.list = [
            Stuff({'label': 'foo'}),
            Stuff({'label': 'bar'}),
            Stuff({'label': 'baz'})
        ]
        instance.import_data({'name': 'changed name', 'list': [Stuff({'label': 'xyz'})]}, recursive=True)
```

**Comment by lkraider** [comment]
Thanks for the test, I've applied a fix. It will be available when I merge #466

**Comment by Guibod** [comment]
You're welcome :)
